nvme-print.c

```diff
@@ -6255,10 +6261,15 @@ void nvme_show_smart_log(struct nvme_smart_log *smart, unsigned int nsid,
 		smart->percent_used);
 	printf("endurance group critical warning summary: %#x\n",
 		smart->endu_grp_crit_warn_sumry);
-	printf("data_units_read				: %s\n",
-		uint128_t_to_string(le128_to_cpu(smart->data_units_read)));
-	printf("data_units_written			: %s\n",
-		uint128_t_to_string(le128_to_cpu(smart->data_units_written)));
+	printf("Data Units Read				: %s %s\n",
+		uint128_t_to_string(le128_to_cpu(smart->data_units_read)),
+		human ?
+		int128_to_si_string(smart->data_units_read, 1000 * 512) : "");
```

Comment:
`data_units_read` is unsigned, so we should have a `uint128_to_si_string` here and not `int128_to_si_string`

Reply:
Actually int128_to_si_string passes `data_units_read ` as __u8 pointer to int128_to_double.
```
char *int128_to_si_string(__u8 *data, __u32 bytes_per_unit)
{
	static char str[40];
	long double val = int128_to_double(data) * bytes_per_unit;
...
```
Then also int128_to_double converts the unsigned pointer values to long double so seems no problem.
```
long double int128_to_double(__u8 *data)
{
	int i;
	long double result = 0;

	for (i = 0; i < 16; i++) {
		result *= 256;
		result += data[15 - i];
	}
	return result;
}
```

Reply:
The data members are either `unsigned int` or `int` (for example `smart->data_units_read`).  They can't be both.  

Isn't  `int128_to_double(data) * bytes_per_unit`  prone to overflow as the multiplication is done as integer type?

Reply:
'smart->data_units_read' is defined as __u8 array as below and `le128_to_cpu` converts it to `nvme_uint128_t`.
```
	__u8			data_units_read[16];
```
`int128_to_si_string` uses existing `int128_to_double` and it converts the array to 'long double'. (existing `int48_to_long` also does convert the 48bit size array to uint64_t.)
Now I am thiking to change the name `int128_to_si_string` to `le128_to_si_string` but seems better to change existing `int128_to_double` and `int48_to_long` names also to `le(num)_to_(type)`.

Reply:
You should really use the `le128_to_cpu` getter to extract the value and pass it into the print function as we do it for `uint128_t_to_string`. Having two different API types for this is not good.